python/pyarrow/pandas_compat.py

```diff
@@ -61,15 +62,8 @@ def get_column_metadata(column, name):
 
     return {
         'name': name,
-        'pandas_type': {
-            'string': 'bytes' if PY2 else 'unicode',
-            'datetime64': (
-                'datetimetz' if hasattr(dtype, 'tz')
-                else 'datetime'
-            ),
-            'integer': str(dtype),
-            'floating': str(dtype),
-        }.get(inferred_dtype, inferred_dtype),
+        'arrow_type': str(arrow_type),
+        'pandas_type': str(pandas_type),
```

Comment:
According to https://github.com/pandas-dev/pandas/blob/master/doc/source/developer.rst we need either `'unicode'` or '`bytes`' for strings. Can this ever be `'categorical'`? May need to expand the unit tests to cover the spec document